scripts/tests/twister_blackbox/test_coverage.py

```diff
@@ -298,3 +312,83 @@ def test_coverage_tool(self, capfd, caplog, test_path, test_platforms, out_path,
         assert str(sys_exit.value) == '0'
 
         assert re.search(expected_content, caplog.text), f'{cov_tool} line not found'
+
+    @pytest.mark.parametrize(
+        'test_path, test_platforms, expected_content',
+        TESTDATA_6,
+        ids=[
+            'missing tool'
+        ]
+    )
+    def test_gcov_tool(self, capfd, caplog, test_path, test_platforms, out_path, expected_content):
+        args = ['--outdir', out_path,'-i', '-T', test_path] + \
+            ['--coverage', '--gcov-tool', TEST_DATA, '-v'] + \
+               [val for pair in zip(
+                   ['-p'] * len(test_platforms), test_platforms
+               ) for val in pair]
+
+        with mock.patch.object(sys, 'argv', [sys.argv[0]] + args), \
+                pytest.raises(SystemExit) as sys_exit:
+            self.loader.exec_module(self.twister_module)
+
+        out, err = capfd.readouterr()
+        sys.stdout.write(out)
+        sys.stderr.write(err)
+
+        assert str(sys_exit.value) == '1'
+        for line in expected_content:
+            result = re.search(line, err)
```

Comment:
You could wrap `line` in `re.escape()` if you do not plan on using any RegEx features. Or even just do `assert line in err`